src/core/Microsoft.PowerToys.Settings.UI.Lib/SettingsUtils.cs

```diff
@@ -46,29 +47,50 @@ public static bool SettingsExists(string powertoy = DefaultModuleName, string fi
             return File.Exists(GetSettingsPath(powertoy, fileName));
         }
 
+        private static readonly JsonSerializerOptions JsonSerializerOptions = new JsonSerializerOptions() { WriteIndented = true };
+
         /// <summary>
         /// Get a Deserialized object of the json settings string.
         /// </summary>
         /// <returns>Deserialized json settings object.</returns>
         public static T GetSettings<T>(string powertoy = DefaultModuleName, string fileName = DefaultFileName)
         {
-            var jsonSettingsString = File.ReadAllText(GetSettingsPath(powertoy, fileName));
+            var jsonSettingsString = File.ReadAllBytes(GetSettingsPath(powertoy, fileName));
             return JsonSerializer.Deserialize<T>(jsonSettingsString);
         }
 
-        // Save settings to a json file.
-        public static void SaveSettings(string jsonSettings, string powertoy = DefaultModuleName, string fileName = DefaultFileName)
+        public static void SaveSettings(string settingsString, string powertoy = DefaultModuleName, string fileName = DefaultFileName)
+        {
+            try
+            {
+                var jsonSettings = Encoding.UTF8.GetBytes(settingsString);
+
+                if (!SettingsFolderExists(powertoy))
+                {
+                    CreateSettingsFolder(powertoy);
+                }
+
+                File.WriteAllBytes(GetSettingsPath(powertoy, fileName), jsonSettings);
+            }
+            catch
+            {
```

Comment:
The current method does have the same generic try/catch logic (introduced with #2129), so i did not change this. But i agree we should explain/log the error. I don't see any logging frameworks added to the Settings projects. Are there any ideas how to handle this? Since it's dotnet core we should base it on Microsoft.Extensions.Logging.Abstractions which is .NET standard compatible, so we can pretty much use it in all our projects.

I'd suggest we handle that in a separate PR?

Reply:
yeah, makes sense to move it into a separate issue